contracts/oracle/DynamicVoltSystemOracle.sol

```diff
@@ -0,0 +1,216 @@
+
+/// @notice contract that linearly interpolates VOLT rate over a year
+/// into the VOLT price after the oracle start time.
+/// The VOLT rate is dynamic and depends on a DynamicVoltRateModel and
+/// the current ratio of liquid reserves in the system.
+/// When PCV allocations change, this contract is notified and the oracle
+/// start time updates, creating a new interpolation over 1 year.
+/// @author Eswak, Elliot Friedman
+contract DynamicVoltSystemOracle is CoreRefV2 {
+    /// ------------- Events ---------------
+
+    /// @notice Event emitted when the Volt system oracle compounds.
+    /// Emits the end time of the period that completed and the new oracle price.
+    event InterestCompounded(
+        uint64 periodStartTime,
+        uint192 periodStartOraclePrice
+    );
+
+    /// @notice Event emitted when the Volt system oracle base rate updates.
+    event BaseRateUpdated(
+        uint256 periodStart,
+        uint256 oldRate,
+        uint256 newRate
+    );
+
+    /// @notice Event emitted when the Volt system oracle actual rate updates.
+    event ActualRateUpdated(
+        uint256 periodStart,
+        uint256 oldRate,
+        uint256 newRate
+    );
+
+    /// @notice Event emitted when reference to the PCV Oracle updates.
+    event PCVOracleUpdated(
+        uint256 blockTime,
+        address oldOracle,
+        address newOracle
+    );
+
+    /// @notice Event emitted when reference to the Rate Model updates.
+    event RateModelUpdated(
+        uint256 blockTime,
+        address oldRateModel,
+        address newRateModel
+    );
+
+    /// ---------- Mutable Variables ----------
+
+    /// @notice Start time at which point interest will start accruing, and the
+    /// current ScalingPriceOracle price will be snapshotted and saved.
+    uint64 public periodStartTime;
+
+    /// @notice Acts as an accumulator for interest earned in previous periods
+    /// returns the oracle price from the end of the last period.
+    uint192 public periodStartOraclePrice;
+
+    /// @notice Current amount that oracle price is inflating by (APR, 18 decimals).
+    /// This is set by governance. This is not actual rate Volt increases by, as it
+    /// does not include the boost eventually returned by the DynamicVoltRateModel.
+    uint256 public baseChangeRate;
+
+    /// @notice Actual amount that oracle price is inflating by (APR, 18 decimals)
+    /// derived from base rate volt price interpolates by this amount.
+    uint256 public actualChangeRate;
+
+    /// @notice Reference to the DynamicVoltRateModel smart contract.
+    address public rateModel;
+
+    /// ---------- Constant Variables ----------
+
+    /// @notice The time frame over which all changes in the APR are applied.
+    /// One year was chosen because this is a long time horizon, allowing for
+    /// better precision in computations, and it is also an intuitive way to
+    /// read the value for humans.
+    uint256 public constant TIMEFRAME = 365.25 days;
```

Comment:
I keep asking this to be changed, but this should really be compounded once per month, that way, if interest hasn't been compounded at month 10, a user isn't paying for unproductive capital.

Reply:
I prefer keeping it 1 year, it makes all other variables more understandable, and the situation where nothing happens for a month doesn't seem credible, let alone for 10 months ? let's discuss it next meeting